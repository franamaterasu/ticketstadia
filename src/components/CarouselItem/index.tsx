import { Fest } from '../../types';

type CarouselItemProps = {
  fest: Fest;
};

const CarouselItem = ({ fest }: CarouselItemProps) => {
  return (
    <div className='relative bg-gray-600 hover:opacity-80'>
      <img
        src={fest.imagen}
        alt={fest.nombre}
        className='rounded-md'
        style={{ height: '524px' }}
      />
      <div className='absolute p-15 left-0 bottom-14 w-full text-center px-5 xl:text-left xl:left-10'>
        <h4 className='text-white font-bold text-5xl lg:mb-5 xl:text-6xl'>
          {fest.nombre}
        </h4>
        <p className='hidden lg:text-white lg:w-1/2 xl:block'>
          {fest.descripcion}
        </p>
      </div>
    </div>
  );
};

export default CarouselItem;
