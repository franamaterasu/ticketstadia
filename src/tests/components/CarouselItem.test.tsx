import { describe, test, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import CarouselItem from '../../components/CarouselItem';

const mockFest = {
  id: 5,
  imagen:
    'https://images.pexels.com/photos/167605/pexels-photo-167605.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
  nombre: 'Indie Waves Festival',
  descripcion:
    'Descubre las últimas bandas y artistas independientes en el escenario indie más grande del año. Indie Waves Festival es el lugar donde las nuevas tendencias musicales toman forma.',
  categoria: 'Indie',
  precio: 65,
  ciudad: 'Barcelona',
  fecha: '2024-09-13',
  destacado: false,
};

describe('<CarouselItem/>', () => {
  test('should match with snapshot', () => {
    const { container } = render(<CarouselItem fest={mockFest} />);

    expect(container).toMatchSnapshot();
  });

  test('should render information about one fest', () => {
    render(<CarouselItem fest={mockFest} />);

    const image = screen.getByRole('img', {
      name: 'Indie Waves Festival',
    });

    const title = screen.getByRole('heading', {
      level: 4,
      name: 'Indie Waves Festival',
    });

    const description = screen.getByText(
      'Descubre las últimas bandas y artistas independientes en el escenario indie más grande del año. Indie Waves Festival es el lugar donde las nuevas tendencias musicales toman forma.'
    );

    expect(image).toBeDefined();
    expect(title).toBeDefined();
    expect(description).toBeDefined();
  });
});
