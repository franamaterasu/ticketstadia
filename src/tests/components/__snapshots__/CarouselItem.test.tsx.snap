// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<CarouselItem/> > should match with snapshot 1`] = `
<div>
  <div
    class="relative bg-gray-600 hover:opacity-80 rounded-md"
  >
    <img
      alt="Indie Waves Festival"
      class="rounded-md"
      src="https://images.pexels.com/photos/167605/pexels-photo-167605.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      style="height: 524px;"
    />
    <div
      class="absolute p-15 left-0 bottom-14 w-full text-center px-5 xl:text-left xl:left-10"
    >
      <h4
        class="text-white font-bold text-5xl lg:mb-5 xl:text-6xl"
      >
        Indie Waves Festival
      </h4>
      <p
        class="hidden lg:text-white lg:w-1/2 xl:block"
      >
        Descubre las últimas bandas y artistas independientes en el escenario indie más grande del año. Indie Waves Festival es el lugar donde las nuevas tendencias musicales toman forma.
      </p>
    </div>
  </div>
</div>
`;
